arbnode/maintenance.go

```diff
@@ -136,41 +141,102 @@ func wentPastTimeOfDay(before time.Time, after time.Time, timeOfDay int) bool {
 	return prevMinutes < dbCompactionMinutes && newMinutes >= dbCompactionMinutes
 }
 
-func (mr *MaintenanceRunner) maybeRunMaintenance(ctx context.Context) time.Duration {
+func (mr *MaintenanceRunner) GetLastMaintenanceTime() (time.Time, error) {
+	milli := mr.lastMaintenance.Load()
+	if milli == 0 {
+		return time.Time{}, errors.New("maintenance running")
```

Comment:
I find it weird that we need to return an error in case maintenance is running.
As a user of this API, I would expect that the time of the last maintenance execution would be returned, even if a maintenance is currently running.

Reply:
I played with things to make it less weird.. though I still like just having one atomic to sync on.